packager/hls/base/media_playlist.cc

```diff
@@ -423,13 +448,15 @@ void MediaPlaylist::AddSegment(const std::string& file_name,
                                          : std::next(iter)->timestamp;
       AddSegmentInfoEntry(file_name, iter->timestamp,
                           next_timestamp - iter->timestamp,
-                          iter->start_byte_offset, iter->size);
+                          iter->start_byte_offset, iter->size, start_timestamp);
```

Comment:
start_timestamp should be incremented by "next_timestamp - iter->timestamp".

Also see the comment below

Reply:
Done.